ranger-http/src/main/java/io/appform/ranger/http/serviceprovider/HttpNodeDataSink.java

```diff
@@ -40,10 +40,12 @@
 public class HttpNodeDataSink<T, S extends HttpRequestDataSerializer<T>> extends HttpNodeDataStoreConnector<T> implements NodeDataSink<T, S> {
 
     private final Service service;
+    private final ObjectMapper mapper;
 
-    public HttpNodeDataSink(Service service, HttpClientConfig config, ObjectMapper mapper, OkHttpClient httpClient) {
-        super(config, mapper, httpClient);
+    public HttpNodeDataSink(Service service, HttpClientConfig config, ObjectMapper mapper, HttpCommunicator<T> httpClient) {
```

Comment:
We can get away with the httpClientConfig also no, if there is a communicator interface already, we need that only to build the OkHttpClient and is not required otherwise, iirc. In that case we can get away with the clientConfig from multiple places.